packages/app/src/runner/event-manager.ts

```diff
@@ -719,15 +719,19 @@ export class EventManager {
     this.ws.off()
   }
 
-  async teardown (state: MobxRunnerStore) {
+  async teardown (state: MobxRunnerStore, isRerun = false) {
     if (!Cypress) {
       return
     }
 
     state.setIsLoading(true)
 
-    // when we are re-running we first
-    // need to stop cypress always
+    if (!isRerun) {
+      // only clear session state when a new spec is selected
+      Cypress.backend('reset:session:state')
```

Comment:
Run mode handles this [here](https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/open_project.ts#L170).

These are the [existing tests](https://github.com/cypress-io/cypress/blob/develop/system-tests/test/session_spec.ts#L142) to verify run mode is working as expected.

Here's a quick link to [`Cypress.backend('reset:session:state')`](https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/socket-base.ts#L460).

Reply:
I'm kind of surprised there isn't more state we want to reset. Now I'm wondering about the basic auth state on visits.